### PR TITLE
Add scroll indicator support in scroll view 

### PR DESCRIPTION
**Pull request by pprabhu-ng**
- Add Scroll Indicator bar support in scroll layer
- Add support for scroll indicator props in scroll view component 
    1) showsHorizontalScrollIndicator
    2) showsVerticalScrollIndicator
    3) indicatorStyle
    4) scrollIndicatorInsets
    5) persistentScrollIndicator ( commented - as parsing of this prop not available in ReactCommon)
- Add support for flashScrollIndicators() methos in scroll view component
- Add scroll indicator support under ENABLE_FEATURE_SCROLL_INDICATOR configuration

### CODE REVIEW

File: rns_shell/compositor/layers/ScrollLayer.cpp [hunk 1]
```diff
@@ -1,17 +1,213 @@
...
+
+ScrollLayer::ScrollBar::ScrollBar() {
+    barColor_.setColor(SK_ColorBLACK);
+    barFrame_ = SkIRect::MakeEmpty();
+    barFrameInsets_ = SkIRect::MakeEmpty();
+}
+
+void ScrollLayer::ScrollBar::showScrollBar(bool show) {
+    if(show) mask_ = LayerPaintInvalidate;
+    else if((!show) && (mask_ == LayerPaintInvalidate)) mask_ = LayerRemoveInvalidate;
```
Comment (pprabhu-ng) [inline rns_shell/compositor/layers/ScrollLayer.cpp:33]: add comment
----
if request to hide scrollbar - set remove mask to update dirty rect 
else not required.

File: rns_shell/compositor/layers/ScrollLayer.cpp [hunk 2]
```diff
@@ -1,17 +1,213 @@
...
+        //If bar frame does not fit in scroll frame,skip drawing scroll bar
+        if((barFrame_.width() > scrollFrame_.width()) || (barFrame_.height() > scrollFrame_.height())) {
+            RNS_LOG_ERROR("Scroll bar size["<< barFrame_.width() <<"," << barFrame_.height() <<"] exceeds scroll layer size[" << scrollFrame_.width() << "," << scrollFrame_.height() <<"]");
+            return;
+        }
+
+        //Bar area can go beyond scroll frame when bar insets are set with negative value
+        //So,we Clip scroll frame area here, to ensure scroll bar does not draw outside scroll frame
+        SkAutoCanvasRestore save(canvas, true);
+        canvas->clipRect(SkRect::Make(scrollFrame_));
```
Comment (pprabhu-ng) [inline rns_shell/compositor/layers/ScrollLayer.cpp:104]: add comment to handle saveLayerAlpha vs clipRect

File: ReactSkia/components/RSkComponentScrollView.h [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
+#include "ReactSkia/sdk/FollyTimer.h"
 
+using namespace rns::sdk;
 namespace facebook {
 namespace react {
 
 #define SCROLL_LAYER_HANDLE static_cast<RnsShell::ScrollLayer*>(layer().get())
 #define SCROLL_LAYER(x) SCROLL_LAYER_HANDLE->x
 #define SCROLLVIEW_DEFAULT_ZOOMSCALE 1
+#define SCROLLBAR_DISPLAY_TIMEOUT 1500
```
Comment (pprabhu-ng) [inline ReactSkia/components/RSkComponentScrollView.h]: rename SCROLLBAR_FADEOUT_TIME

File: ReactSkia/components/RSkComponentScrollView.h [hunk 4]
```diff
@@ -58,7 +62,14 @@ class RSkComponentScrollView final : public RSkComponent {
 
  private:
   bool scrollEnabled_{true};
+  bool isHorizontalScroll_{false};
+  bool showHorizontalScrollIndicator_{false};
+  bool showVerticalScrollIndicator_{false};
+  bool showScrollIndicator_{false};
```
Comment (pprabhu-ng) [inline ReactSkia/components/RSkComponentScrollView.h]: rename to drawScrollIndicator

File: ReactSkia/components/RSkComponentScrollView.cpp [hunk 5]
```diff
@@ -21,22 +21,86 @@ RSkComponentScrollView::RSkComponentScrollView(const ShadowView &shadowView)
...
 
   snapToOffsets_.clear();
-  for(auto &offset : scrollViewProps.snapToOffsets) {
+  for(auto &offset : newScrollViewProps.snapToOffsets) {
     snapToOffsets_.push_back(static_cast<int>(offset));
   }
   sort(snapToOffsets_.begin(),snapToOffsets_.end());
 
-  return RnsShell::LayerInvalidateNone;
+  RnsShell::ScrollLayer* scrollLayer= SCROLL_LAYER_HANDLE;
```
Comment (pprabhu-ng) [inline ReactSkia/components/RSkComponentScrollView.cpp:73]: add scrollbar code in macro 
- ENABLE_FEATURE_SCROLL_INDICATOR

**Review by munezbn — APPROVED** [review]

**Review by chatrapathi06 — APPROVED** [review]

**Pull request merged**